File: 05c-working-with-heterogenous-topic-avro-idl-and-springboot/src/spring-boot-kafka-producer-avro-idl-heterogenous/src/main/java/com/trivadis/kafkaws/springbootkafkaproducer/TopicCreator.java
package com.trivadis.kafkaws.springbootkafkaproducer;

import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.kafka.config.TopicBuilder;
import org.springframework.stereotype.Component;

@Component
public class TopicCreator {
    @Value(value = "${topic.name}")
    private String testTopic;

    @Value(value = "${topic.partitions}")
    private Integer testTopicPartitions;

    @Value(value = "${topic.replication-factor}")
    private short testTopicReplicationFactor;

    @Bean
    public NewTopic testTopic() {
        return TopicBuilder.name(testTopic)
                        .partitions(testTopicPartitions)
                        .replicas(testTopicReplicationFactor)
                        .build();
    }
}

File: 06b-producing-consuming-kafka-with-kotlin-springboot/src/spring-boot-kotlin-kafka-producer/src/main/kotlin/com/trivadis/kafkaws/springbootkotlinkafkaproducer/TopicCreator.kt
package com.trivadis.kafkaws.springbootkotlinkafkaproducer

import org.apache.kafka.clients.admin.NewTopic
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.kafka.config.TopicBuilder
import org.springframework.stereotype.Component

@Component
class TopicCreator {

    @Value("\${topic.name}")
    private lateinit var kafkaTopic: String

    @Value("\${topic.partitions}")
    private var partitions: Int = 0

    @Value("\${topic.replication-factor}")
    private var replicationFactor: Int = 0

    @Bean
    fun topic(): NewTopic? {
        return TopicBuilder.name(kafkaTopic)
            .partitions(partitions)
            .replicas(replicationFactor)
            .build()
    }
}